### PR TITLE
fix flake8 for new plugins

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #5743 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Comment by Borda** [comment]
Seems some tests are failing...

**Review by Borda — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -223,23 +217,6 @@ def validation_step_end(self, output):
         """
         return self.training_type_plugin.validation_step_end(output)
 
-    def predict(self, args):
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:226]: @tchaton we had multiple declarations of this method inside a single class and after removing the second one TPU tests are failing... mind have look at it...

Comment (tchaton) [reply pytorch_lightning/accelerators/accelerator.py:226]: Yes, I will

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 2]
```diff
@@ -263,7 +263,7 @@ def distributed_sampler_kwargs(self):
         distributed_sampler_kwargs = dict(num_replicas=self.world_size, rank=self.global_rank)
         return distributed_sampler_kwargs
 
-    def init_optimizers(self, trainer: "Trainer", model: LightningModule) -> Tuple[List, List, List]:
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/deepspeed.py:266]: @Borda I had to remove these trainer references, because flake8 complains Trainer is not imported. But if we import it, we get cyclic import errors. So I don't know a better way.

**Comment by Borda** [comment]
@awaelchli any idea why TPU is failing?
cc: @tchaton

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 3]
```diff
@@ -206,7 +206,7 @@ def post_dispatch(self) -> None:
         # restore main state with best weights
         best_path = self.mp_queue.get()
         last_path = self.mp_queue.get()
-        results = self.mp_queue.get()
+        self._results = self.mp_queue.get()
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:209]: does it mean that we have not used it in past? @tchaton

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:209]: Yes, it was None by default I guess.

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:209]: Just need to update the test
